### DISCUSSION TITLE
Battery Usage While Hassio Ecoflow Running

### DISCUSSION THREAD

**Discussion by mattwells** [vwt12eh8/hassio-ecoflow#35] [General]
I have noticed that my battery is running down a lot faster now that I am reading the information from the batttery, quite significantly infact. Has anyone else noticed this?

**Comment by jiriteach** [comment]
> I have noticed that my battery is running down a lot faster now that I am reading the information from the batttery, quite significantly infact. Has anyone else noticed this?

Yes I was about to create a similar post. As of the newer versions I am seeing decent battery drain!

My Rivers are on bypass but I do have my battery level set at 85% since they are bypass. They are charged to 100 and seeing drain. There's nothing else to drain the battery since it's on bypass. 

![image](https://user-images.githubusercontent.com/2116446/183270191-a8071ffa-f6a6-45ed-98d5-8ddb422be79a.png)

**Comment by skpow55** [comment]
Hmmm.  I have not really noticed on my Delta Pro  but I am not running 
much of a load right now and plugged in to AC.  It would make sense that 
the radio/wifi power consumption would increase on the tx side as you 
are now at least doubling the transmission packets  by sending the data 
to both the Ecoflow cloud server to be consumed by the EF phone app also 
also to your HA EF integration via local wifi network connection.   but 
power consumption for radio/wifi use should be in the ma range so you 
would think it would be nominal even doubled.

Also consider the idle/active state power consumption be the electronics 
in the EF itself without any loads.

 From EF Support:

     When the AC Inverter is ON, the power loss will be 30W- 40W, 1% per 
hour.

     When the AC Inverter is OFF, the power loss will be 10W - 20W, 0.5% 
per hour.


I guess one could run the two test below to confirm additional power 
draw from the HA integration.

     1. HA EF integration disabled and EF in standby/idle mode with no 
load draw

     2. HA EF integration enabled and EF in standby/idle mode with no 
load draw

**Comment by vwt12eh8** [comment]
Enabling integration may increase power consumption, but I don't think it will be significant.

The process of getting the status from the device is basically done by push. The timing is controlled by the device side, and it seems that it is not possible to adjust the notification interval from here.

There is an entity that needs polling partially, but I do not think that power consumption increases so much because it is the same 30 seconds as the default polling interval of HomeAssistant.

As a countermeasure, there is a method of connecting only for a short time at a fixed interval without always connecting, but this loses the user experience because real-time update becomes impossible, and I do not know whether there is a power consumption improvement effect.

**Comment by skpow55** [comment]
Ahhh.  So if the EF is doing a interval data push only  then there is no 
additional radio/wifi packets being transmitted as you are doing a 
listen on the EF port only.   I think the radio/wifi chip is on a esp32 
mc board. Do you know if this is the case?

**Comment by mattwells** [comment]
Having done some testing over the last few days I have come to the conclusion that it is a difference in solar along with myself being a little less mindful of my energy usage while on battery. I have done 2 evenings without monitoring and 2 with and they both seems to be about the same.

I have now got hold of a few smart plugs  that work with HA and have set one up to automatically turn mains power on and disable charging before the battery goes to low.

**Comment by mattwells** [comment]
After my Delta Pro refused to charge higher than 79% yesterday either from the mains or solar. 

What I think has happened is that over time the calibration of the battery has fallen out of sync as I don't normally let it discharge all the way and I think turning off the AC Charging while passing power through to ensure that my devices continue to be powered. 

What I have done to try and fix this is fully discharged the battery to 0% and today fully charged it up. I think this has probably fixed it as it took quite a long time to charge from 99% to 100%.

**Reply by JasonMM2000** [reply]
On the "official ecoflow club" group on facebook they say to power it off by holding the button till it says off.  Then hold the button for 10 seconds and you should see it jump to new numbers around the 7 or 8 second mark.  That will initiate a recalibration.  The next firmware update is supposed to address this issue but no idea when it will arrive.  

I've done it a bunch and it always gives a different number.

**Reply by Ne0-Hack3r** [reply]
Top-Balance + Recalibration Procedure: Charge to 100%, disconnect from AC input/loads, turn OFF, long press holding 15 seconds to ON (BMS Reset), then turn OFF again, and leave OFF for at least 5 hours, then discharge to ZERO before recharging to you normal recharge limit, then put hack into normal service/cycle. This works for Delta Pro. Not sure about other models...
